src/MongoDB/Cursor.c

```diff
@@ -502,6 +516,9 @@ void php_phongo_cursor_init_ce(INIT_FUNC_ARGS) /* {{{ */
 	PHONGO_CE_DISABLE_SERIALIZATION(php_phongo_cursor_ce);
 	php_phongo_cursor_ce->get_iterator = php_phongo_cursor_get_iterator;
 
+#if PHP_VERSION_ID >= 80000
```

Comment:
why not implementing IteratorAggregate on all PHP versions ?

Reply:
Since the internal iterator only exists on PHP 8, we decided to only implement this on PHP 8 for the time being. You can find some more motivation in the original ticket: https://jira.mongodb.org/browse/PHPC-1690.

I may end up revisiting this sooner than later depending on how PHPLIB reacts to these changes.

Reply:
hmm, I forgot that the internal iterator is a PHP 8 feature. That's a good reason then :smile: